src/nvim/mouse.c

```diff
@@ -1110,7 +1110,7 @@ int jump_to_mouse(int flags, bool *inclusive, int which_button)
     ? wp->w_winbar_height != 0
     : false;
 
-  on_statuscol = (grid == (col < win_col_off(wp)))
+  on_statuscol = (!on_status_line && grid == DEFAULT_GRID_HANDLE && col < win_col_off(wp))
```

Comment:
A check for winbar may also be needed

Reply:
Doesn't seem necessary. Winbar clicks do not register as statuscolumn clicks currently. I guess because `grid == DEFAULT_GRID_HANDLE` is already false when the mouse is on winbar.

Reply:
Added a regression test for both.

Reply:
Checking for `DEFAULT_GRID_HANDLE` doesn't make sense at all for `on_statuscol`.

Reply:
Alright, I'll take your word for it. I'm not familiar with the ui grids. Seems to work as expected for both winbar and statusline when omitting the grid handle check as well.

Reply:
The value of `on_statuscol` isn't used when `on_winbar` is true, but when I tested this with GDB I can see that with mouse at top-left corner of window both `on_statuscol` and `on_winbar` are true.